dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/jsl/impl/JobModelSerializerImpl.java

```diff
@@ -16,61 +16,56 @@
  */
 package com.ibm.jbatch.container.jsl.impl;
 
+import java.io.ByteArrayOutputStream;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
 import javax.xml.bind.JAXBContext;
 import javax.xml.bind.JAXBElement;
 import javax.xml.bind.Marshaller;
 import javax.xml.namespace.QName;
 
-
-
-
 import com.ibm.jbatch.container.jsl.JSLValidationEventHandler;
 import com.ibm.jbatch.container.jsl.ModelSerializer;
-import com.ibm.jbatch.container.jsl.ValidatorHelper;
 import com.ibm.jbatch.jsl.model.JSLJob;
 
-import java.io.ByteArrayOutputStream;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
-
 public class JobModelSerializerImpl implements ModelSerializer<JSLJob> {
 
-	@Override
-	public String serializeModel(JSLJob model) {
-		
-		final JSLJob finalModel = model;
-		String serializedModel = AccessController.doPrivileged(
-		    	 new PrivilegedAction<String>() {
-		              public String run() {
-		            	  return marshalJSLJob(finalModel);
-		              }
-		          });
-		
-		return serializedModel;
-	}
+    @Override
+    public String serializeModel(JSLJob model) {
+
+        final JSLJob finalModel = model;
+        String serializedModel = AccessController.doPrivileged(
+                                                               new PrivilegedAction<String>() {
+                                                                   @Override
+                                                                   public String run() {
+                                                                       return marshalJSLJob(finalModel);
+                                                                   }
+                                                               });
+
+        return serializedModel;
+    }
 
     private String marshalJSLJob(JSLJob job) {
-    	String resultXML = null;
-    	JSLValidationEventHandler handler = new JSLValidationEventHandler();
-    	try {
-    		JAXBContext ctx = JAXBContext.newInstance("com.ibm.jbatch.jsl.model");
-    		Marshaller m = ctx.createMarshaller();
-    		m.setSchema(ValidatorHelper.getXJCLSchema());
-    		m.setEventHandler(handler);
-    		ByteArrayOutputStream baos = new ByteArrayOutputStream();
-    		//m.marshal(job, baos);
-    		/*
-    		 * from scott: 
-    		 */
-    		m.marshal( new JAXBElement(
-    				new QName("http://xmlns.jcp.org/xml/ns/javaee","job"), JSLJob.class, job ), baos);
-    		resultXML = baos.toString();
-    	}
-    	catch(Exception e){
-    		throw new RuntimeException("Exception while marshalling JSLJob", e);
-    	}
-    	
-    	return resultXML;
+        String resultXML = null;
+        JSLValidationEventHandler handler = new JSLValidationEventHandler();
+        try {
+            ClassLoader currentClassLoader = JSLJob.class.getClassLoader();
+            JAXBContext ctx = JAXBContext.newInstance("com.ibm.jbatch.jsl.model", currentClassLoader);
```

Comment:
Did this line really matter (passing the classloader explicitly) or was it maybe something you changed trying to work through the schema validation problem?

Reply:
Yup, had to add it to avoid a classNotFound exception.  Had to fix that to get far enough to *have* the validation problem.

Reply:
OK, I see as we moved to Open Liberty the model was separated out into a **com.ibm.jbatch.jsl.model** bundle/project, which it hadn't been previously.  Makes sense.